osu.Server.Spectator/Hubs/Multiplayer/MultiplayerHub.cs

```diff
@@ -167,11 +167,11 @@ public async Task<MultiplayerRoom> JoinRoomWithPassword(long roomId, string pass
 
             try
             {
-                await sharedInterop.AddUserToRoomAsync(Context.GetUserId(), roomId, password);
+                // Run in background so we don't hold locks on user/room states.
+                _ = sharedInterop.AddUserToRoomAsync(Context.GetUserId(), roomId, password);
             }
-            catch (Exception ex)
+            catch
```

Comment:
I'd probably leave an inline comment inside the catch block here (and the other one) that any failures are logged internally to `SharedInterop` because this looks very not good at face value.